### ISSUE TITLE
The fusion of `Conv2d`, `BatchNorm2d`, and `ReLU` failed

### ISSUE THREAD

**Issue by AnonymousChaseDreamer** [pytorch/pytorch#112820]
### 🐛 Describe the bug

When I try to use `torch.ao.quantization.fuse_modules` to fuse operators, it fails. Specifically, `Conv2d`, `BatchNorm2d`, and `ReLU` can not be fused as expected. After operator fusion, they are fused into `ConvReLU2d`. However, in the official document, the fused operator should be `ConvBatchNormReLU2d`. 

When applying these fusion rules in the real world. It may result in the wrong output.

Here is the code to reproduce the bugs
```python
from torch.ao.quantization.fuse_modules import fuse_modules
import torch.nn as nn

class M(nn.Module):
    def __init__(self):
        super(M, self).__init__()
        self.conv1 = nn.Conv2d(3, 64, 3, 1, 1)
        self.bn1 = nn.BatchNorm2d(64)
        self.relu1 = nn.ReLU()
        self.conv2 = nn.Conv2d(64, 64, 3, 1, 1)
        self.bn2 = nn.BatchNorm2d(64)
        self.relu2 = nn.ReLU()

    def forward(self, x):
        x = self.conv1(x)
        x = self.bn1(x)
        x = self.relu1(x)
        x = self.conv2(x)
        x = self.bn2(x)
        x = self.relu2(x)
        return x

m = M().eval()

modules_to_fuse = ['conv1', 'bn1', 'relu1']

print(f'before operator fusion: \n{m}\n')
fused_m = fuse_modules(m, modules_to_fuse)
print(f'after operator fusion: \n{fused_m}\n')
```

Here is the output
```
before operator fusion: 
M(
  (conv1): Conv2d(3, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
  (relu1): ReLU()
  (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
  (relu2): ReLU()
)

after operator fusion: 
M(
  (conv1): ConvReLU2d(
    (0): Conv2d(3, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
    (1): ReLU()
  )
  (bn1): Identity()
  (relu1): Identity()
  (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
  (relu2): ReLU()
)
```

Can you help me confirm this bug? Thx in advance!

### Versions

Collecting environment information...
PyTorch version: 2.1.0
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.8.2
Libc version: glibc-2.27

Python version: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-4.15.0-213-generic-x86_64-with-glibc2.27
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: Tesla V100-SXM2-32GB
GPU 1: Tesla V100-SXM2-32GB
GPU 2: Tesla V100-SXM2-32GB
GPU 3: Tesla V100-SXM2-32GB

Nvidia driver version: 470.182.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.3
[pip3] onnx==1.13.1
[pip3] onnxoptimizer==0.3.13
[pip3] onnxruntime==1.16.1
[pip3] torch==2.1.0
[pip3] torchaudio==2.1.0
[pip3] torchvision==0.16.0
[conda] Could not collect

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

**Comment by jerryzh168** [comment]
replied in https://github.com/pytorch/pytorch/issues/112579#issuecomment-1804307017

let's track things there, I'll close this one since it's the same issue
